include/my_sys.h

```diff
@@ -174,6 +174,8 @@ extern void my_free(void *ptr);
 extern void *my_memdup(PSI_memory_key key, const void *from,size_t length,myf MyFlags);
 extern char *my_strdup(PSI_memory_key key, const char *from,myf MyFlags);
 extern char *my_strndup(PSI_memory_key key, const char *from, size_t length, myf MyFlags);
+extern void *aligned_malloc(size_t size, size_t align);
+extern void aligned_free(void *ptr);
```

Comment:
this should take PSI_memory_key as an argument too

Reply:
aligned_malloc is currently only used in InnoDB and memory allocation is not instrumented as per existing semantics.
New usage in table-cache-instance too doesn't demand memory instrumentation.
So if we continue to retain existing semantics (for the scope of the PR) then the said param is not needed.
If we want all the aligned_malloc calls to be memory instrumented then semantics needs to be worked out and accordingly, more keys could be defined (more feasible under separate PR given the unrelated scope).

Reply:
Supporting `PERFORMANCE_SCHEMA` instrumentation does not seem to be doable without some storage overhead. I was initially thinking about something that would incur no runtime overhead for the current use cases (passing `key=PSI_NOT_INSTRUMENTED`, like this:
```c++
inline void *my_aligned_alloc(PSI_memory_key key, size_t size, size_t align)
{
  void *buf= IF_WIN(_aligned_malloc,aligned_alloc)(size, align);
  if (key && buf) m_key= PSI_CALL_memory_alloc(key, size, &m_owner);
  return buf;
}

inline void my_aligned_free(void *buf, size_t size)
{
  IF_WIN(_aligned_free,free)(buf);
  PSI_CALL_memory_free(m_key, m_owner);
}
```
This is based on `my_malloc()` and `my_free()`. The `m_key` and `m_owner` are placeholders for something, and freeing memory would only need `size` so that it can be passed to the instrumentation layer.

If we bloated each aligned allocation with 24 bytes or `my_memory_header` like `my_malloc()` does and `my_free()` expects, that would seem to defeat the very purpose of an aligned allocator. In the bad old days, there was some code like this in InnoDB:
```c
	buf2 = ut_malloc(2 * UNIV_PAGE_SIZE);
	/* Align the memory for file i/o if we might have O_DIRECT set */
	page = ut_align(buf2, UNIV_PAGE_SIZE);
```
If `PERFORMANCE_SCHEMA` instrumentation of aligned memory allocation is truly needed, it can be implemented separately, using an interface where the metadata is stored separately from the aligned buffer.